Library/Homebrew/cask/lib/hbc/cask_loader.rb

```diff
@@ -31,6 +31,12 @@ def initialize(path)
 
         @token = path.basename(".rb").to_s
         @path = path
+
+        @tap ||= if @path == CaskLoader.default_path(@token)
```

Comment:
@reitermarkus Thanks for your work on #2965 , which makes this part unnecessary. I've rebased and removed the unneeded code, so now this doesn't need to touch `CaskLoader` or the `Cask` constructor at all.